### PR TITLE
fix: [CDS-34621]: Serverless Infratab changes

### PR DESCRIPTION
**Pull request by satyam-harness**
##### Jira Links:
https://harness.atlassian.net/browse/CDS-34621
https://harness.atlassian.net/browse/CDS-35689

##### Screenshots:

Infrastructure Tab for serverless deployement (selected ServerlessAwsLambda as deployment type)
Screenshot 2022-03-31 at 6 31 18 PM

AWS is selected.
Screenshot 2022-03-31 at 6 31 31 PM

We can see serverless deployment infrastructure specific fields in the runpipeline form
Screenshot 2022-03-31 at 6 34 17 PM

We can see serverless deployment infrastructure specific fields in the crate inputset form
Screenshot 2022-03-31 at 6 34 31 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/ServerlessAwsLambdaSpec.test.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        onUpdate={onUpdateHandler}
+        ref={ref}
+      />
+    )
+
+    await act(async () => {
+      const stageInput = container.querySelector('[placeholder="cd.steps.serverless.stagePlaceholder"]')
+      fireEvent.change(stageInput!, { target: { value: 'stage changed' } })
+
+      // TODO: add other fields
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/ServerlessAwsLambdaSpec.test.tsx:196]: Address this TODO in next PR.

File: src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/ServerlessAwsLambdaSpec.test.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+      await ref.current?.submitForm()
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/ServerlessAwsLambdaSpec.test.tsx:198]: May be, we do not have Submit button under this component. If it is there then click on button and submit form.

File: src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/mock/ConnectorsResponse.mock.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+export const ConnectorsResponse = {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/mock/ConnectorsResponse.mock.ts:8]: Create just one file for Connector responses under mock folder and keep all connector responses in that file.

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx [hunk 4]
```diff
@@ -122,17 +150,23 @@ export default function DeployInfraSpecifications(props: React.PropsWithChildren
...
   React.useEffect(() => {
-    const type = stage?.stage?.spec?.infrastructure?.infrastructureDefinition?.type
-    setSelectedDeploymentType(type)
-    const initialInfraDefValues = getInfrastructureDefaultValue(stage, type)
+    const selectedType = stage?.stage?.spec?.serviceConfig?.serviceDefinition?.type
+    const infrastructureType = selectedType
+      ? deploymentTypeInfraTypeMap[selectedType]
+      : InfraDeploymentType.KubernetesDirect
+    setselectedInfrastructureType(infrastructureType)
+    const initialInfraDefValues = getInfrastructureDefaultValue(stage, infrastructureType)
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx:165]: Earlier, **stage?.stage?.spec?.infrastructure?.infrastructureDefinition?.type** was passed to **getInfrastructureDefaultValue** as second argument.
Now, you are passing **deploymentTypeInfraTypeMap[stage?.stage?.spec?.serviceConfig?.serviceDefinition?.type]**
Hope that is expected

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/deployInfraHelper.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    delete stageData.stage.spec.infrastructure.infrastructureDefinition.provisioner
+    isChanged = true
+  }
+
+  return isChanged
+}
+
+export const getInfrastructureDefaultValue = (
+  stageData: StageElementWrapper | undefined,
+  deploymentType: string | undefined
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/deployInfraHelper.tsx:46]: Change in next PR - This variable name should be **infrastructureType**

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx [hunk 6]
```diff
@@ -95,7 +95,7 @@ const getCGTypes = (
 
 const getServerlessDeploymentTypes = (
   getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string,
-  SERVERLESS_SUPPORT = false
+  SERVERLESS_SUPPORT = true
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:98]: **Too dangerous** - Please revert this change. I had a comment for this earlier as well,

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/__tests__/__snapshots__/DeployStageSetupShell.test.tsx.snap [hunk 7]
```diff
@@ -1232,6 +1232,85 @@ exports[`DeployStageSetupShell tests opens services tab by default 1`] = `
                             </div>
                           </span>
                         </span>
+                        <span
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/__tests__/__snapshots__/DeployStageSetupShell.test.tsx.snap:1235]: This change should not appear if there is no corresponding change in test file.

File: src/modules/75-cd/strings/strings.en.yaml [hunk 8]
```diff
@@ -85,6 +87,9 @@ pipelineSteps:
   infraTab:
     deploymentType: Infrastructure type selection is required
     clusterError: No cluster found with given Connector
+    regionError: No region found with given Connector
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/strings/strings.en.yaml:90]: Is this change intended ? regionError and stageError

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]
Need to revert default value of feature flag. We can address other comments as well along with it.

File: src/modules/75-cd/strings/strings.en.yaml (hunk 8 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/strings/strings.en.yaml:90]: Fixed.

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/__tests__/__snapshots__/DeployStageSetupShell.test.tsx.snap (hunk 7 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/__tests__/__snapshots__/DeployStageSetupShell.test.tsx.snap:1235]: Fixed.

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx (hunk 6 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:98]: Fixed.

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/deployInfraHelper.tsx (hunk 5 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/deployInfraHelper.tsx:46]: Fixed,

File: src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/mock/ConnectorsResponse.mock.ts (hunk 3 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/ServerlessAWSLambda/__test__/mock/ConnectorsResponse.mock.ts:8]: Fixed.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**